EIPS/eip-4972.md

```diff
@@ -0,0 +1,137 @@
+
+## Motivation
+A popular way to onboard Web2 users to the Web3 world is custody. However, current custody models have severe drawbacks. Considering the following widely adopted custody models:
+1. The custodian uses one EOA/CA to hold the assets of all users. This is not compatible with on-chain social protocols since all user activities are off-chain.
+2. One EOA per user. The social identity is not portable, which means there is no way for users to migrate their social identity across different custody platforms.
+3. One CA (e.g. Gnosis Safe) per user. The one time deployment cost is super high and the user experience is not good.
+
+To solve all these problems, this ERC proposes a new type of account - name-owned account (NOA). An NOA has
+1. a name for social identity; and
+2. an address derived from the name to receive tokens; and
+3. an owner/owners of the name that is able to transfer the token.
+
+The name should be human-readable and can be easily recognized socially. An example is the username of a centralized platform such as FB, Twitter. The name-derived address (NDA) is a normal Ethereum address that should not collide with the existing addresses of EOA/CA. Since we cannot use NDA as msg.sender, the right to transfer the tokens of the NDA is controlled by the owner/owners of the name. The name to owner/owners mapping is managed by an on-chain name service, and the owner/owners are EOA/CA, which can be the addresses of 3-rd custodians (e.g. FB) or self-custodian. By changing the owner of the name to the EOA of the user (can be done by requesting the custodian), the NDA becomes self-custodian, and no one should be able to transfer the assets unless the approved by the self-custodian user. 
+
+Using NOA as social identity instead of EOA/CA brings huge benefits for users:
+- **Easy Web2 user onboarding**. We are providing standard Web2 user experiences with human readable names and 3rd-party custody. Every user of a centralized platform can immediately have a NOA by using the username as the name of NOA custodied by the platform.
+- **On-chain states**. All user states are on-chain even under custody so it’s 100% compatible with social protocols.
+- **Portable Account**. A NOA can be easily ported to different custody platforms by changing the owner.
+- **Flexible Account Management**. We can use one EOA/CA to control any number of NOAs.
+
+## Specification
+### Name Representation
+A name is represented by a bytes array which is ABI encoded. It **MAY** contain metadata of the name such as the name service the name belongs to.  Examples of the name are "vitalik.eth", "vitalik@0x00000000000C2E074eC69A0dFb2997BA6C7d2e1e", or "qizhou.fb".
+
+## Interface
+### INameOwnedAccount
+```
+interface INameOwnedAccount {
+    /// @notice This function resolves the _name to its derived address
+    /// @dev The implementation SHOULD avoid collision between name 
+    /// derived address and EOA/CA
```

Comment:
Do you have a recommendation for how to avoid such a collision?

Reply:
A recommendation can be found in "Reference Implementation" section.